impala/datadog_checks/impala/metrics_daemon.py

```diff
@@ -0,0 +1,388 @@
+    # daemon
+    "impala_server_backend_num_queries_executed": "daemon.num_queries_executed",
+    "impala_server_backend_num_queries_executing": "daemon.num_queries_executing",
+    "impala_server_num_files_open_for_insert": "daemon.num_files_open_for_insert",
+    "impala_server_num_fragments": "daemon.num_fragments",
+    "impala_server_num_fragments_in_flight": "daemon.num_fragments_in_flight",
+    "impala_server_num_open_beeswax_sessions": "daemon.num_open_beeswax_sessions",
+    "impala_server_num_open_hiveserver2_sessions": "daemon.num_open_hiveserver2_sessions",
+    "impala_server_num_queries": "daemon.num_queries",
+    "impala_server_num_queries_expired": "daemon.num_queries_expired",
+    "impala_server_num_queries_registered": "daemon.num_queries_registered",
+    "impala_server_num_queries_spilled": "daemon.num_queries_spilled",
+    "impala_server_num_sessions_expired": "daemon.num_sessions_expired",
+    "impala_server_ddl_durations_ms": "daemon.ddl_durations_ms",
+    "impala_server_query_durations_ms": "daemon.query_durations_ms",
+    "impala_server_scan_ranges_num_missing_volume_id": "daemon.scan_ranges_num_missing_volume_id",
+    "impala_server_scan_ranges": "daemon.scan_ranges",
+    "impala_server_hedged_read_ops": "daemon.hedged_read_ops",
+    "impala_server_hedged_read_ops_win": "daemon.hedged_read_ops.win",
+    "impala_total_senders_blocked_on_recvr_creation": "daemon.total_senders_blocked_on_recvr_creation",
+    "impala_total_senders_timedout_waiting_for_recvr_creation": "daemon.total_senders_timedout_waiting_for_recvr_creation",
+    "impala_request_pool_service_resolve_pool_duration_ms_total": counter(
+        "daemon.request_pool_service_resolve_pool_duration"
+    ),
+    "impala_senders_blocked_on_recvr_creation": "daemon.senders_blocked_on_recvr_creation",
+    # statestore_subscriber
+    "impala_statestore_subscriber_heartbeat_interval_time_total": counter(
+        "daemon.statestore_subscriber.heartbeat_interval_time"
+    ),
+    "impala_statestore_subscriber_last_recovery_duration": "daemon.statestore_subscriber.last_recovery_duration",
+    "impala_statestore_subscriber_num_connection_failures": "daemon.statestore_subscriber.num_connection_failures",
+    "impala_statestore_subscriber_statestore_client_cache_clients_in_use": "daemon.statestore_subscriber.statestore_client_cache.clients_in_use",
+    "impala_statestore_subscriber_statestore_client_cache_total_clients": "daemon.statestore_subscriber.statestore_client_cache.total_clients",
+    "impala_statestore_subscriber_topic_update_duration_total": counter(
+        "daemon.statestore_subscriber.topic.update_duration"
+    ),
+    "impala_statestore_subscriber_topic_update_interval_time_total": counter(
+        "daemon.statestore_subscriber.topic.update_interval_time"
+    ),
+}
+
+DAEMON_METRICS_WITH_LABEL_IN_NAME = {
+    "^impala_server_io_mgr_queue_([0-9]+)_(.+)$": {
+        "label_name": "id",
```

Comment:
Are we sure queue_id is not high cardinality? how many queues are possible?

Reply:
Up to 24 with 5 metrics each

Reply:
Oh that's fine then, thanks!